### PR TITLE
Yang model changes for Sequential IDF isolation

### PR DESCRIPTION
**Pull request by tjchadaga**
#### Why I did it
To define a field to track isolation state for the IDF to which the T2 device belongs

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
Added yang model definition, unit tests, sample config and documentation for the table

#### How to verify it
Validated config tree generation using "pyang -Vf tree -p /usr/local/share/yang/modules/ietf ./yang-models/sonic-voq-inband-interface.yang"

Built the below python-wheels to validate unit tests and other changes
target/python-wheels/bullseye/sonic_yang_mgmt-1.0-py3-none-any.whl
target/python-wheels/bullseye/sonic_yang_models-1.0-py3-none-any.whl
target/python-wheels/bullseye/sonic_config_engine-1.0-py3-none-any.whl

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ]  202205 (chassis)
- [ ]

### CODE REVIEW

File: src/sonic-yang-models/yang-models/sonic-bgp-device-global.yang [hunk 1]
```diff
@@ -21,12 +21,26 @@ module sonic-bgp-device-global {
...
             container STATE {
                 description "BGP device-specific global data";
-                leaf tsa_enabled {
+                leaf tsa_enabled {                    
                     type boolean;
                     default "false";
                     description
                     "When set to true, Traffic is shifted away (TSA), i.e, BGP routes are not advertised to neighboring routers";
                 }
+                leaf idf_isolation_state {
```
Comment (qiluo-msft) [inline src/sonic-yang-models/yang-models/sonic-bgp-device-global.yang:30]: >idf_isolation_state [](http://example.com/codeflow?start=21&length=19)

Is there any limitation of this field on M0/T0/T1 devices? If no, could we generalize the isolation for all roles, not to mention IDF?

Comment (tjchadaga) [reply src/sonic-yang-models/yang-models/sonic-bgp-device-global.yang:30]: IDF isolation is only supported on T2 and the config will not have any effect on T0/T1 devices.

Comment (qiluo-msft) [reply src/sonic-yang-models/yang-models/sonic-bgp-device-global.yang:30]: Then can we enforce a value for M0/T0/T1? The yang model could help the enforcement.

Comment (tjchadaga) [reply src/sonic-yang-models/yang-models/sonic-bgp-device-global.yang:30]: Currently, the field is initialized to "unisolated" on all devices (including M0/T0/T1) in init_cfg.json.j2.  Changing this configuration through CLI/Script is blocked on non-T2 devices. Is that sufficient?

File: src/sonic-yang-models/yang-models/sonic-bgp-device-global.yang [hunk 2]
```diff
@@ -21,12 +21,26 @@ module sonic-bgp-device-global {
         container BGP_DEVICE_GLOBAL {
             container STATE {
                 description "BGP device-specific global data";
-                leaf tsa_enabled {
+                leaf tsa_enabled {
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-bgp-device-global.yang:24]: Please remove space from end of line.

File: src/sonic-yang-models/yang-models/sonic-bgp-device-global.yang [hunk 3]
```diff
@@ -21,12 +21,26 @@ module sonic-bgp-device-global {
...
+                leaf tsa_enabled {                    
                     type boolean;
                     default "false";
                     description
                     "When set to true, Traffic is shifted away (TSA), i.e, BGP routes are not advertised to neighboring routers";
                 }
+                leaf idf_isolation_state {                    
+                    type enumeration {
+                        enum isolated_no_export {
+                            description "IDF isolated using no-export community tag";
```
Comment (xincunli-sonic) [inline src/sonic-yang-models/yang-models/sonic-bgp-device-global.yang:33]: >IDF isolated using no-export community tag [](http://example.com/codeflow?start=41&length=42)

Can we add DC size - Medium/Large

File: src/sonic-yang-models/yang-models/sonic-bgp-device-global.yang [hunk 4]
```diff
@@ -21,12 +21,26 @@ module sonic-bgp-device-global {
...
+                        }
+                        enum isolated_withdraw_all {
+                            description "IDF isolated by withdrawing routes";
```
Comment (xincunli-sonic) [inline src/sonic-yang-models/yang-models/sonic-bgp-device-global.yang:36]: >IDF isolated by withdrawing routes [](http://example.com/codeflow?start=41&length=34)

Same, also add DC size - Small/X-Small ?

File: src/sonic-yang-models/yang-models/sonic-bgp-device-global.yang (hunk 3 above)
Comment (tjchadaga) [reply src/sonic-yang-models/yang-models/sonic-bgp-device-global.yang:33]: This field is just meant to indicate that the isolation is to be done by advertising routes tagged with no-export BGP community and is independent of any DC size.

File: src/sonic-yang-models/yang-models/sonic-bgp-device-global.yang (hunk 4 above)
Comment (tjchadaga) [reply src/sonic-yang-models/yang-models/sonic-bgp-device-global.yang:36]: This field is just meant to indicate that the isolation is to be done by withdrawing routes and is independent of any DC size.

File: src/sonic-yang-models/yang-models/sonic-bgp-device-global.yang (hunk 2 above)
Comment (tjchadaga) [reply src/sonic-yang-models/yang-models/sonic-bgp-device-global.yang]: done

File: src/sonic-yang-models/yang-models/sonic-bgp-device-global.yang (hunk 1 above)
Comment (qiluo-msft) [reply src/sonic-yang-models/yang-models/sonic-bgp-device-global.yang]: What is the backend behavior if a T0 got config which is not unisolated? will it fail or ignore?

If the backend will treat this as a failure, we can also hardening yang models, so when ConfigDB is validated by yang models, the failure could be detected earlier, and stop the configuration process immediately.

Comment (tjchadaga) [reply src/sonic-yang-models/yang-models/sonic-bgp-device-global.yang]: Only T2 will handle this config change. T0/T1 will ignore it

**Review by arlakshm — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]
LGTM. Please check with other active reviewers.

**Pull request merged**